accepted/00062-hub-xmlrpc-api.md

```diff
@@ -0,0 +1,136 @@
+- Feature Name: hub_xmlrpc_api
+- Start Date: 2019-10-10
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Uyuni Server has an XMLRPC API that allows access to most of its functionality.
+
+With the Hub architecture, there will be multiple Servers - this RFC details a new XMLRPC API for the Hub to access Server functionality.
+
+See the [Hub general RFC](accepted/00060-hub-general.md) for an introduction to the Hub project.
+
+# Motivation
+[motivation]: #motivation
+
+End users might not want to access single Servers to automate operations across their infrastructure, so a Hub XMLRPC API endpoint would be useful to make third party scripting and integration easier.
+
+From the requirements:
+
+> Hub implements API endpoints to act on SUMA slave Servers
+>
+>    Main expected use cases:
+>        Content management (eg. channel cloning)
+>        Automated patching
+
+# Detailed design
+[design]: #detailed-design
+
+### Basic funtionality
+The Hub is a Server, so as far as it and its minions (Servers) are concerned, the Hub XMLRPC API work as of today.
+Content management APIs will work as of today. Inter-Server Synchronization (ISS) will be used to push managed content from Hub to Servers.
+
+### New service and endpoint
+A new XMLRPC API endpoint will be created, implemented by a new service called the "XMLRPC Gateway API" (simply called "Gateway" from now on in this document). Technology-wise:
+  * Implementation will be in Python based on top of Tornado and [tornado-xmlrpc](https://pypi.org/project/tornado-xmlrpc/) library
```

Comment:
Why implement from scratch with Tornado instead of using an existing API gateway?
E. g. [Graviteer](https://gravitee.io/products/apim/) (first page in Google) already implements [caching](https://docs.gravitee.io/apim_policies_cache.html), [rate limiting](https://docs.gravitee.io/apim_policies_rate_limiting.html), can [serve XMLRPC as REST](https://docs.gravitee.io/apim_policies_rest2soap.html) APIs, does [RBAC](https://docs.gravitee.io/apim_policies_role_based_access_control.html), [API keys](https://docs.gravitee.io/apim_policies_apikey.html) to limit what API calls are available to what users, analytics, load balancing, etc.

Other API gateways may provide similar or better features: [WSO2 API Management](https://wso2.com/api-management/), [Tyk](https://tyk.io/features/api-gateway/), [APIman](http://www.apiman.io/), [KrakenD CE](https://github.com/devopsfaith/krakend-ce), etc.

Reply:
As per our last meeting, I will be following up on this possibility through an add-on RFC.